diff --git a/grub-core/commands/add.c b/grub-core/commands/add.c
new file mode 100644
--- a/grub-core/commands/add.c	2025-07-10 17:59:04.100597101 +0000
+++ b/grub-core/commands/add.c	2025-07-10 18:09:08.147205933 +0000
@@ -0,0 +1,28 @@
+#include <grub/command.h>
+#include <grub/err.h>
+
+static grub_err_t
+grub_cmd_add (grub_command_t cmd __attribute__ ((unused)), int argc, char **argv)
+{
+  if (argc != 3)
+    return grub_error (GRUB_ERR_BAD_ARGUMENT, "usage: add <num1> <num2>");
+
+  long n1 = grub_strtoul (argv[1], 0, 0);
+  long n2 = grub_strtoul (argv[2], 0, 0);
+
+  grub_printf ("%ld\n", n1 + n2);
+  return GRUB_ERR_NONE;
+}
+
+static grub_command_t cmd;
+
+GRUB_MOD_INIT(add)
+{
+  cmd = grub_register_command ("add", grub_cmd_add, 0, "Add two numbers.");
+}
+
+GRUB_MOD_FINI(add)
+{
+  grub_unregister_command (cmd);
+}
+

